app/controllers/coronavirus_form/session_controller.rb

```diff
@@ -3,6 +3,6 @@
 class CoronavirusForm::SessionController < ApplicationController
   def delete
     reset_session
-    redirect_to "/"
+    redirect_to params[:r] || "/"
```

Comment:
We'll need to make sure we don't have an [Open Redirect](https://cwe.mitre.org/data/definitions/601.html) here.

Reply:
ah, hummm right.

So i could post on twitter https://gov.uk/clear-session?r=https://huws-evil-site-of-doom.com
and i'd get all the nice meta data stuff from our site but fling them onto my evil phising site?

Reply:
humm ok not good, will perhaps just make it a switch and hard code in a single redirect location

Reply:
like internal = gov.uk root, external = bbc.co.uk or something?

Reply:
@richardTowers had a second pass, let me know if that might work instead?

Reply:
Yep, your understanding of the issue is correct, and your new implementation looks good to me.

These issues sometimes pop up with login flows, where it's like `/login?on_success_redirect=some-phishing-site.com`, which is particularly nasty. It would be a bit less bad than that in this case, but still worth avoiding.

Reply:
Right, good catch. One for my TIL list!